autoload/asyncomplete/sources/flow.vim

```diff
@@ -66,6 +71,27 @@ function! asyncomplete#sources#flow#get_source_options(opts)
        \ })
 endfunction
 
+let s:cached_flowbin_path_by_dir = {} " dir: <path to flow>
+
+function! s:resolve_flowbin_path(file, fallback)
+    let l:dir = fnamemodify(a:file, ':h')
+    " cap the cache so it won't grow unlimited and cause a memory leak
```

Comment:
memory leaks sounds like a bug in vim or this plugin. Might be change to something like out of memory errors? and also add a note that we should ideally have something like LRU cache.